### DISCUSSION TITLE
MUIでの実装が難しいところをCSSinJSで実装したい

### DISCUSSION THREAD

**Discussion by nto300002** [if-mentor/next_fire_todo_5#59] [Ideas]
issue #16 
にてMUIを利用してスタイルを当てていたのですが、textareaのサイズの固定ができない（※[こちら](https://www-creators.com/archives/2496)のようなことができない）ので、MUIで実装が難しいところは個人的にCSSinJSで実装したいと思うのですがよろしいでしょうか？

**Comment by Yukitkym** [comment]
@nto300002 MUIについて私が詳しくないので合っているか分からないのですが、sx Propにスタイルを当てていく方法では難しそうですか？
https://qiita.com/cieloazul310/items/d630da98439c89d773ba#%E4%B8%80%E8%88%AC%E3%81%AE-html-%E8%A6%81%E7%B4%A0%E3%81%AE%E3%82%B9%E3%82%BF%E3%82%A4%E3%83%AA%E3%83%B3%E3%82%B0

**Reply by nto300002** [reply]
htmlでtextareaをそのまま書くとweb上からでもサイズを自由に変えられてしまうのですが、それを固定するためにcssでresize: none;を設定することができます。
しかし、MUIで使えるTextareaAutosizeというコンポーネントは、sxではresize: none;を設定できないみたいなんですよね
他にも@mui/systemをつかってスタイルを設定してもresize: none;だけは設定できなかったので、困っている状況です

**Reply by Junkichi89** [reply]
@nto300002 

CSS in JSを使うのは、問題ないと思います！
使うのであれば、チームで統一する必要があるので、MTGで相談の上、ルールは決めた方がいいと思います。

ただ、特に何も考えずに、resizeさせないだけであれば、インラインスタイルで記述するだけでいいように思いますが、どうでしょうか？

また、textareaのサイズを固定するというのは、要件に含まれていないと思いますが、
固定しないといけないと思うのはどういった理由があるのでしょうか？
もし、デザイン通りにするためということであれば、そこまで細かくデザインと全く一緒になってなくても大丈夫ですよ🤡
多少はオリジナルが入っても許容の範囲内かと思います。

**Reply by takosan8** [reply]
@nto300002 
自分はCSS in JS使うのは問題ないと思います。
ただ適用範囲を考慮すると、インラインスタイルで記述する方が最小限でいいかなと感じました！

**Reply by nto300002** [reply]
すみません、普通にスタイル適用できました。回答していただいてありがとうございます！

**Reply by nto300002** [reply]
なにか問題が発生するまでは今後もMUIで頑張ってみます
